projects/js-toolkit/packages/liferay-cli/src/build.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import {format} from '@liferay/js-toolkit-core';
+import child_process from 'child_process';
+import resolve from 'resolve';
+
+const {error, print} = format;
+
+export default async function build(): Promise<void> {
+	const pkgJson = projectRequire('./package.json');
+
+	if (!pkgJson.dependencies) {
+		print(error`Project has no dependencies: cannot find target platform`);
+
+		return;
+	}
+
+	const targetPlatformName = Object.keys(pkgJson.dependencies).find(
+		(packageName) => {
+			const packageJson = projectRequire(`${packageName}/package.json`);
+
+			if (
+				packageJson.dependencies &&
+				packageJson.dependencies['liferay-npm-bundler']
+			) {
+				return true;
+			}
+
+			return false;
+		}
+	);
```

Comment:
Yeah, that's [what we agreed in the issue](https://github.com/liferay/liferay-frontend-projects/issues/542#issuecomment-844109982), but we can still change it.

I think it makes sense, though, because having a target platform without the bundler wouldn't work because you wouldn't be able to build the project. And there's no other scenario in which pulling the bundler with a preset makes sense, because prior to target platforms, it was the bundler who used the preset, so we are doing an inversion of control now, thus there won't be instances of this inversion of control in the past.

In any case, we can expand this logic in the future if we need to. For example, we can use keywords to tag target platforms that don't include the bundler, use a npm scope, or anything else we may define, ...

So, unless we can think of a case where this detection logic would pose a problem, I think we can go with it for now, as we agreed.

Reply:
I agree that looking for `liferay-npm-bundler` it's not the cleanest check we can make, but on the other hand, `liferay-npm-bundler` is not going to change (only for bundler v3, but that's incompatible with this presets) and it's a npm package name, so something (almost) set in stone, as a URL or a UUID.

Think of it as a tagging mechanism. And if we need to make target platforms that don't include `liferay-npm-bundler` as a dependency in the future, we can expand this logic, as I said...

Reply:
Related to https://github.com/liferay/liferay-frontend-projects/pull/576